ocaml/xenops/domain.ml

```diff
@@ -211,15 +213,26 @@ let make ~xc ~xs vm_info uuid =
 			(* create cpu and memory directory with read only perms *)
 			List.iter (fun dir ->
 				let ent = sprintf "%s/%s" dom_path dir in
-				t.Xst.mkdir ent;
-				t.Xst.setperms ent roperm
+				mkdirperms t ent roperm
 			) [ "cpu"; "memory" ];
+
+
+			let mksubdirs base dirs perms =
+				List.iter (fun dir ->
```

Comment:
This is a minor point, but I think `List.iter (fun dir -> mkdirperms t (base^"/"^dir) perms) dir` would be as readable. One could also use `sprintf`. But I would not mind if you keep it the way it is.